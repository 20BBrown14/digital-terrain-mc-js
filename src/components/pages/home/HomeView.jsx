import React from 'react';
import PropTypes from 'prop-types';
import { Typography, Carousel, Space } from 'antd';
import './Home.css';
import spawnFallbackImage from '../../../assets/Spawn_Fallback.png';
import {
  NK_ABOUT_US,
  NK_GALLERY,
  NK_MAP,
  NK_APPLY,
} from '../../../constants/navKeys';

const propTypes = {
  /* Function to navigate to new page */
  navigateToNewPage: PropTypes.func.isRequired,
  /* Images to display in featured carousel */
  carouselImages: PropTypes.arrayOf(PropTypes.object).isRequired,
  /* Whether carousel images load failed */
  isCarouselLoadingFailed: PropTypes.bool.isRequired,
};

/**
 * Home page view.
 */
const HomeView = (props) => {
  const {
    navigateToNewPage,
    carouselImages,
    isCarouselLoadingFailed,
  } = props;

  const pandaLink = (
    <button
      id="pandaLink"
      className="home-page-link-button"
      type="button"
      onClick={() => { navigateToNewPage(NK_ABOUT_US); }}
      onKeyPress={() => { navigateToNewPage(NK_ABOUT_US); }}
    >
      Panda_Brah
    </button>
  );
  const dmLink = (
    <button
      id="dmLink"
      className="home-page-link-button"
      type="button"
      onClick={() => { navigateToNewPage(NK_ABOUT_US); }}
      onKeyPress={() => { navigateToNewPage(NK_ABOUT_US); }}
    >
      DMDF_
    </button>
  );
  const galleryLink = (
    <button
      id="galleryLink"
      className="home-page-link-button"
      type="button"
      onClick={() => { navigateToNewPage(NK_GALLERY); }}
      onKeyPress={() => { navigateToNewPage(NK_GALLERY); }}
    >
      gallery
    </button>
  );
  const mapLink = (
    <button
      id="mapLink"
      className="home-page-link-button"
      type="button"
      onClick={() => { navigateToNewPage(NK_MAP); }}
      onKeyPress={() => { navigateToNewPage(NK_MAP); }}
    >
      map
    </button>
  );
  const applyLink = (
    <button
      id="applyLink"
      className="home-page-link-button"
      type="button"
      onClick={() => { navigateToNewPage(NK_APPLY); }}
      onKeyPress={() => { navigateToNewPage(NK_APPLY); }}
    >
      apply here
    </button>
  );

  return (
    <div className="home-page-content">
      <Typography.Title className="home-page-title">
        Welcome to Digital Terrain
      </Typography.Title>
      <Space direction="vertical">
        <Typography.Text className="home-page-subtitle">
          {'Digital Terrain is an SMP server inspired by Hermitcraft. The server is based on a great community, trust,\
          and fun. We aspire to have a playerbase of all play styles that work together in harmony to build great\
          things and have an awesome time. The server is owned by '}
          {pandaLink}
          {' and '}
          {dmLink}
          {'. Alongside them we have a team of mods who help make sure everything is running smoothly.'}
        </Typography.Text>
        <Typography.Text className="home-page-subtitle">
          {'Be sure to checkout our '}
          {galleryLink}
          {' and '}
          {mapLink}
          {" to get a feel for the server; hopefully you'll find it's a good place for you!\
          If you decide to, you can "}
          {applyLink}
          {' to be a part of our server. Hope to play with you soon!'}
        </Typography.Text>
      </Space>
      <Carousel
        autoplay
        className="home-page-carousel"
      >
        {!isCarouselLoadingFailed
          ? carouselImages.map((image) => (
            <img src={image.address} alt="loading..." key={`image-${image.id}`} />
          ))
          : <img src={spawnFallbackImage} alt="loading..." key="home-page-image-fallback" />}
      </Carousel>
    </div>
  );
};

HomeView.propTypes = propTypes;

export default HomeView;
