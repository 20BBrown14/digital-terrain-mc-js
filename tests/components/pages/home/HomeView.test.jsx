import React from 'react';
import HomeView from '../../../../src/components/pages/home/HomeView';

describe('HomeView', () => {
  it('renders a default view', () => {
    const view = render(<HomeView />);
    expect(view).toMatchSnapshot();
  });
});
