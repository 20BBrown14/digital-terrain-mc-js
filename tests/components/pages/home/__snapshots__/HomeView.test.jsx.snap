// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView renders a default view 1`] = `
<div
  class="home-page-content"
>
  <h1
    class="ant-typography home-page-title"
  >
    Welcome to Digital Terrain
  </h1>
  <div
    class="ant-space ant-space-vertical"
  >
    <div
      class="ant-space-item"
      style="margin-bottom:8px"
    >
      <span
        class="ant-typography home-page-subtitle"
      >
        Digital Terrain is a SMP server inspired by Hermitcraft. The server is based on a great community, trust,          and fun. We aspire to have a playerbase of all play styles that work together in harmony to build great          things and have an awesome time. The server is owned by 
        <button
          class="home-page-link-button"
          id="pandaLink"
          type="button"
        >
          Panda_Brah
        </button>
         and 
        <button
          class="home-page-link-button"
          id="dmLink"
          type="button"
        >
          DMDF_
        </button>
        . Alongside them we have a team of mods who help make sure everything is running smoothly.
      </span>
    </div>
    <div
      class="ant-space-item"
    >
      <span
        class="ant-typography home-page-subtitle"
      >
        Be sure to checkout our 
        <button
          class="home-page-link-button"
          id="galleryLink"
          type="button"
        >
          gallery
        </button>
         and 
        <button
          class="home-page-link-button"
          id="mapLink"
          type="button"
        >
          map
        </button>
         to get a feel for the server; hopefully you'll find it's a good place for you!          If you decide to, you can 
        <button
          class="home-page-link-button"
          id="applyLink"
          type="button"
        >
          apply here
        </button>
         to be a part of our server. Hope to play with you soon!
      </span>
    </div>
  </div>
  <div
    class="ant-carousel"
  >
    <div
      class="slick-slider home-page-carousel slick-initialized"
    >
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="width:0%;left:NaN%"
        />
      </div>
    </div>
  </div>
</div>
`;
